tests/tests/integration_tests.rs

```diff
@@ -477,10 +468,11 @@ async fn ws_server_subscribe_with_stream() {
 	// Sub1 is still in business
 	assert_eq!(sub1.next().await.unwrap().unwrap(), 3);
 
-	// terminate connection.
-	// TODO: (dp) removing the drop changes nothing. Why is that?
-	drop(client);
+	// We expect the stream to run five times and should be getting 5 values on `rx`.
+	// We've read three, so two more expected and then the channel is closed.
+	assert_eq!(Some(()), rx.next().await, "subscription stream should be terminated after the client was dropped");
```

Comment:
sorry David, `rx` is not coupled to the actual client subscription just to for us to know if test context that the subscription was terminated if that makes sense.

perhaps we should do something different if the code is so damn hard to understand